pecan/lang/pecan_ast/arith.py

```diff
@@ -85,47 +52,43 @@ def __repr__(self):
         return '({} - {})'.format(self.a, self.b)
 
     def evaluate(self, prog):
-        # if type(self.a) is IntConst and type(self.b) is IntConst:
-        #     return IntConst(self.a.evaluate_int() - self.b.evaluate_int()).evaluate(prog)
+        if self.is_int:
+            return IntConst(self.evaluate_int(prog))
```

Comment:
This should also evaluate the int constant right, to generate the corresponding automata?

Reply:
addressed in commit ed38512